Experimental/DistributedAuthoritySample/Assets/Scripts/Player/ThirdPersonRBController.cs

```diff
@@ -0,0 +1,227 @@
+    if (Mathf.Abs(moveInput.x) > 0.01f)  // Only rotate if moveInput.x is significantly non-zero
+    {
+        Quaternion deltaRotation = Quaternion.Euler(new Vector3(0, moveInput.x * turnSpeed * 30f, 0) * Time.deltaTime);
+        rb.MoveRotation(rb.rotation * deltaRotation);
+    }
+
+    // jumping
+    if (Input.GetButtonDown("Jump") && grounded)
+    {
+        animator.SetTrigger("Jump");
+        rb.AddForce(new Vector3(0, 1, 0) * jumpForce * 100f);
+    }
+
+    rb.AddForce(new Vector3(0, -1, 0) * gravityScale);
+
+    // grabbing item
+    if (Input.GetButtonDown("Fire1") && currentPickupItem == null)
+    {
+        // Find closest pickup item within range
+        Collider[] hitColliders = Physics.OverlapBox(new Vector3(transform.position.x, transform.position.y, transform.position.z + 0.5f), new Vector3(pickupRange / 2, pickupRange / 2, pickupRange / 2));
+        foreach (var hitCollider in hitColliders)
+        {
+            if (hitCollider.CompareTag("PickUpItem"))
+            {
+                // Calculate direction to the item
+                Vector3 directionToItem = (hitCollider.transform.position - transform.position).normalized;
+                float dotProduct = Vector3.Dot(transform.forward, directionToItem);
+
+                // Check if the item is within the field of view
+                if (dotProduct > pickupAngleThreshold)
+                {
+                    // Execute logic for picking up the item
+                    currentPickupItem = hitCollider.gameObject;
+                    // Rotate the player to face the item smoothly
+                    StartCoroutine(SmoothLookAt(currentPickupItem.transform));
+                    animator.SetTrigger("Pickup");
+                    break; // Exit the loop after picking up the first item
+                }
+            }
+        }
+    }
+
+    // throwing or dropping item
+    if (currentPickupItem != null && Input.GetButtonDown("Fire2"))
+    {
+        StartCoroutine(ThrowOrDropItem());
+    }
+
+    // Apply a dead zone to the angular velocity to zero out small values
+    if (Mathf.Abs(rb.angularVelocity.y) < 0.7f)
+    {
+        rb.angularVelocity = new Vector3(rb.angularVelocity.x, 0, rb.angularVelocity.z);
+    }
+}
+
+    void OnDrawGizmos()
+    {
+        // Visualize the pickup range in the editor
+        Gizmos.color = Color.yellow;
+        Gizmos.DrawWireCube(new Vector3(transform.position.x, transform.position.y, transform.position.z + 0.5f), new Vector3(pickupRange, pickupRange, pickupRange));
+    }
+
+    public void PickupAction()
+    {
+        // Create FixedJoint and connect it to the player's hand
+        currentPickupItem.GetComponent<Rigidbody>().detectCollisions = false;
+        currentPickupItem.transform.position = pickupLocChild.transform.position;
+        currentPickupItem.transform.rotation = pickupLocChild.transform.rotation;
+        pickupLocfixedJoint.connectedBody = currentPickupItem.GetComponent<Rigidbody>();
```

Comment:
@NoelStephensUnity 

I'd get your eyes on this, since this seems to be the inverse of the approach that NetworkRigidbody takes, no?

In this approach, the Avatar character has a Rigidbody at the root, and a child bone in the rig with a Rigidbody & FixedJoint component.

It's worthwhile comparing it to the approach [here](https://github.com/Unity-Technologies/com.unity.multiplayer.samples.bitesize/blob/feat/distributed-authority-sample/pickup/Experimental/DistributedAuthoritySample/Assets/Scripts/Player/PickupAction.cs), done through NetworkRigidbody.

Could you instead have a NetworkRigidbody on a child GameObject, and attach the item pickup object to that? I'll experiment with that!